### ISSUE TITLE
Support for ES6 Proxy results in always new instances in instance() call

### ISSUE THREAD

**Issue by thomashilzendegen** [NagRock/ts-mockito#80]
The `instance()` call for a mock returns always a new `Proxy` class instance (when supported). In that case e.g. a `verify()` does not recognise the parameter when an instance of a mock is used.

This is a different behaviour compared to the result of an `instance()` call when `Proxy` is not supported by the browser.

A simple (somehow constructed) example:

```
const fooMock = mock(Foo);
const fooInstance = instance(fooMock);

// ... some code later

when(fooMock.clone(instance(fooMock))).thenReturn(fooInstance); 
/*
the "when()" does not match because a new Proxy class will be
returned from "instance()" for each call (works in an older
browser not supporting Proxy)
*/
fooInstance.clone(fooInstance);
```

Although the example is quite constructed, the `instance()` method should not change its behaviour depending on the environment.

**Comment by NagRock** [comment]
Hi, @thomashilzendegen. Thanks for catching this issue.
Should be fixed in 2.2.8v https://github.com/NagRock/ts-mockito/releases/tag/v2.2.8

**Comment by thomashilzendegen** [comment]
@NagRock Thank you very much!
